mmdet/structures/bbox/base_bbox.py

```diff
@@ -0,0 +1,412 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from abc import ABCMeta, abstractclassmethod, abstractproperty
+from typing import List, Optional, Sequence, Tuple, TypeVar, Union
+
+import numpy as np
+import torch
+from torch import Tensor
+
+T = TypeVar('T')
+DeviceType = Union[str, torch.device]
+IndexType = Union[slice, int, list, torch.LongTensor, torch.cuda.LongTensor,
+                  torch.BoolTensor, torch.cuda.BoolTensor]
+
+
+class BaseInstanceBoxes(metaclass=ABCMeta):
```

Comment:
I think the box type should also overwrite some basic operators such as "+ - * / @"

Reply:
It needs more discussion. I consider basic operators usually don't have specific physical meaning for boxes. They will treat boxes as a tensor, which have different effects for different box modes.